src/libtelio.udl

```diff
@@ -75,7 +75,7 @@ interface TelioError {
 /// Possible adapters.
 enum TelioAdapterType {
     /// Userland rust implementation.
-    "BoringTun",
+    "NepTUN",
```

Comment:
Damn this is incompatible as well, as all app team code will stop compiling once we merge this. 

In practice, here we have two choices:
* Bump major version, communicate to all relevant teams what has changed and what they need to do. And go through that.
* Keep backwards compatibility.

IMHO Moving towards `NepTUN` alone is not really enough to justify major version release (especially while it is almost unmodified boringtun under-the-hood). Therefore IMHO I believe it would be nicer to have "BoringTun" enum variant which would get mapped to NepTUN.

And once we accumulate some more reasons to do major version bump -> we can do the change in one go, as it will necessitate to go through all the non-backwards-compatibility hoops anyways.